deep_quoridor/src/renderers.py

```diff
@@ -28,20 +28,42 @@ class NoneRenderer(Renderer):
 
 
 class ResultsRenderer(Renderer):
-    def start_game(self, game, agent1: Optional[Agent] = None, agent2: Optional[Agent] = None):
-        self.match = f"{agent1.__class__.__name__} vs {agent2.__class__.__name__}"
+    def start_game(self, game, agent1: Agent, agent2: Agent):
+        self.match = f"{agent1.name()} vs {agent2.name()}"
 
-    def end_game(self, game, step):
-        print(f"Game over! {self.match}: {game.winner()} won in {step} steps")
+    def end_game(self, game, result: GameResult):
+        print(f"{self.match}: {result.winner} won in {result.steps} steps")
 
+    def end_arena(self, game, results: list[GameResult]):
+        won = {}
+        won_as_p1 = {}
+        for r in results:
+            for player in [r.player1, r.player2]:
+                if player not in won:
+                    won[player] = 0
+                    won_as_p1[player] = 0
 
-class TextRenderer(Renderer):
-    def start_game(self, game, agent1: Optional[Agent] = None, agent2: Optional[Agent] = None):
+            if r.winner == r.player1:
+                won[r.player1] += 1
+                won_as_p1[r.player1] += 1
+            else:
+                won[r.player2] += 1
+
+        print("\n")
+        print("Player          | Wins   | Wins P1 ")
```

Comment:
Hmm ... I have the impression that more usual metrics are "win %" or at least "wins / total", otherwise it doesn't say much.
Also, I think a table showing pairwise results are more typical. I'm not an expert, but I suggest doing a quick search to see what's usual.

At the very minimum please include wins out of how many.

Reply:
Ok, the new version will show for example:

```
+------------+--------+--------+------------+--------+
|  P1 \ P2   | random | simple | simpleplus | Total  |
+------------+--------+--------+------------+--------+
|   random   |   -    |  60%   |    40%     |  50%   |
|   simple   |  80%   |   -    |     0%     |  40%   |
| simpleplus |  40%   |  80%   |     -      |  60%   |
|   ======   | ====== | ====== |   ======   | ====== |
|   Total    |  60%   |  70%   |    20%     |  50%   |
+------------+--------+--------+------------+--------+
```

Reply:
Actually, that format doesn't work well.
I'm switching to something like this in another PR:

```
+---------------+--------+--------+------------+-------+
|     Player    | random | simple | simpleplus | Total |
+---------------+--------+--------+------------+-------+
|   P1 random   |   -    |  20%   |    60%     |  40%  |
|   P2 random   |   -    |  20%   |    20%     |  20%  |
|   P1 simple   |  80%   |   -    |    80%     |  80%  |
|   P2 simple   |  80%   |   -    |    20%     |  50%  |
| P1 simpleplus |  80%   |  80%   |     -      |  80%  |
| P2 simpleplus |  40%   |  20%   |     -      |  30%  |
+---------------+--------+--------+------------+-------+
```